BitwardenShared/Core/Vault/Services/SyncService.swift

```diff
@@ -256,6 +270,11 @@ extension DefaultSyncService {
         try await policyService.replacePolicies(response.policies, userId: userId)
         try await stateService.setLastSyncTime(timeProvider.presentTime, userId: userId)
         try await checkVaultTimeoutPolicy()
+
+        if try await keyConnectorService.userNeedsMigration(),
+           let organization = try await keyConnectorService.getManagingOrganization() {
+            await delegate?.removeMasterPassword(organizationName: organization.name)
+        }
```

Comment:
🤔 I see this is being done on any sync which might be problematic:
1. Could it misbehave when sync is called from an extension? E.g. in Fido2 flow [here](https://github.com/bitwarden/ios/blob/main/BitwardenShared/Core/Vault/Services/Fido2CredentialStoreService.swift#L45)
2. If the user is currently in a modal view and a notification comes which triggers sync like `syncVault`, would it work as expected or the modal will block it?
3. Same as 2 but instead of a modal view, an alert error is currently presented in the screen.

Reply:
Good questions! I added a check for not showing this screen in the extensions. This will display over a modal/alert since we dismiss any modal views before navigating to the auth coordinator.

https://github.com/bitwarden/ios/blob/e1bb0b73895115a404c02062a19dad117ecc7603/BitwardenShared/UI/Platform/Application/Utilities/RootViewController.swift#L19-L21


https://github.com/user-attachments/assets/e48c30ab-dba2-47d2-a793-8a24d45a90c2

Reply:
Thanks 🎉